File: Object/C++/animals/cat.cpp
#include <stdio.h>
#include "cat.h"

Cat::Cat(std::string myName, std::string ownName): Animal(number++, myName, ownName){}
Cat::Cat(std::string myName): Animal(number++, myName){}

int Cat::CatCount() {
    return number;
}

void Cat::Identify() const {
    printf("A cat.\n");
    Animal::Identify();
}

void Cat::MakeNoise() const {
    printf("Meow\n");
}

File: Object/Java/animals/Cat.java
public class Cat extends Animal {
    private static int number = 0;

    public Cat(String anName) {
        super(++number, anName);
    }

    public Cat(String anName, String anOwner) {
        super(++number, anName, anOwner);
    }

    public void MakeNoise() {
        System.out.println("Meow.");
    }

    public void Identify() {
        System.out.println("I am a cat.");
        super.Identify();
    }

    public static int CatCount() {
        return number;
    }
}
